packages/@lwc/engine-core/src/framework/check-version-mismatch.ts

```diff
@@ -41,7 +41,7 @@ export function checkVersionMismatch(
         const version = versionMatcher[1];
         const [major, minor] = version.split('.');
         const [expectedMajor, expectedMinor] = LWC_VERSION.split('.');
-        if (major !== expectedMajor || minor !== expectedMinor) {
+        if (major !== expectedMajor || minor !== expectedMinor || version !== LWC_VERSION) {
```

Comment:
```suggestion
        if (version !== LWC_VERSION) {
```
If we're comparing the full strings, then there's no need to check the substrings.

Reply:
@wjhsf  I thought of the same but issue name is "to warn for patch difference as well" so I have not removed existing logic. I will update the above changes. Thanks!!